### PR TITLE
#152220239 User Signin

### PR DESCRIPTION
**Pull request by ayodelevm**
#### What does this PR do?
It implement User sigin feature

#### Description of Task to be completed?
- refactored authenticateUser function to loginUser function
- Added isAuthenticated function for token verification
- Added and refactored function to validate payload
- Refactored createUser method and route
- Added sigin route and middleware

#### What are the relevant pivotal tracker stories?
#152220239

### CODE REVIEW

File: src/server/routes/authRoutes.js [hunk 1]
```diff
@@ -2,9 +2,15 @@ import express from 'express';
...
+authRouter.post(
+  '/api/v1/user/signup',
   AuthController.createUser(),
-  AuthController.authenticateUser()
-]);
+);
+
+authRouter.post(
+  '/api/v1/user/signin',
+  AuthController.isAuthenticated(),
```
Comment (tomipaul) [inline src/server/routes/authRoutes.js]: @ayodelevm The `isAuthenticated` controller mounted here suggests that the token is sent along in the request header.

File: src/server/controllers/AuthController.js [hunk 2]
```diff
@@ -23,36 +27,80 @@ class AuthController {
...
   /**
-   * Authenticate a user with username and password
+   * Authenticates user token
+   * @method
+   * @memberof AuthController
+   * @static
+   * @return {function} Express middleware function that calls the
+   * next method after authentication has been completed
+   */
+  static isAuthenticated() {
```
Comment (tomipaul) [inline src/server/controllers/AuthController.js]: @ayodelevm I'm guessing there's a mixup here. This controller is not needed for signup and signin functionalities. This should come in subsequently when user is making a request to access a resource.
And there's the `getClientToken` and `authorizeUser` controllers for that already.

File: src/server/routes/authRoutes.js (hunk 1 above)
Comment (ayodelevm) [reply src/server/routes/authRoutes.js]: Oh... my mistake, will fix that right away

File: src/server/controllers/AuthController.js (hunk 2 above)
Comment (ayodelevm) [reply src/server/controllers/AuthController.js]: didn't realize that has been added. Will fix it.

**Comment by ayodelevm** [comment]
done

**Pull request merged**